clean_all.sh

```diff
@@ -2,9 +2,11 @@
 
 # SPDX-License-Identifier: Apache-2.0
 
+# Cleans all build artifacts, except for the environment and toolchains.
+# If you want to clean the environment (should be rarely needed), remove the env/build directories.
+# E.g. rm -rf env/build third_party/tt-mlir/env/build
+
 rm -rf build
```

Comment:
What do you think about having two scripts:
- clean_build.sh - Cleans FFE, MLIR and Metal build
- clean_all.sh - Runs clean_build.sh + Removes env

Reply:
I am ok with renaming `clean_all.sh` -> `clean_build.sh`... But I would avoid adding a script for cleaning up build + env, since the need for it should be extremely rare.